### PR TITLE
support setting DJANGO_PROJECT_FOLDER via --project-root-path

### PR DESCRIPTION
**Pull request by linuxmaniac**
settings file can be located in a subdirectory of the project

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -58,6 +57,7 @@ Add the migration linter your ``INSTALLED_APPS``:
 ``--database DATABASE``                            Specify the database for which to generate the SQL. Defaults to *default*.
 ``--cache-path PATH``                              specify a directory that should be used to store cache-files in.
 ``--no-cache``                                     Don't use a cache.
+``--project-root-path=DJANGO_PROJECT_FOLDER``      An absolute or relative path to the django project.
```
Comment (flixx) [inline README.rst]: ``--project-root-path DJANGO_PROJECT_FOLDER``  to stay consistant ;) (with space, not =)

File: README.rst [hunk 2]
```diff
@@ -58,6 +57,7 @@ Add the migration linter your ``INSTALLED_APPS``:
 ``--database DATABASE``                            Specify the database for which to generate the SQL. Defaults to *default*.
 ``--cache-path PATH``                              specify a directory that should be used to store cache-files in.
 ``--no-cache``                                     Don't use a cache.
+``--project-root-path DJANGO_PROJECT_FOLDER``      An absolute or relative path to the django project.
```
Comment (linuxmaniac) [inline README.rst]: done

File: README.rst [hunk 3]
```diff
@@ -43,12 +43,11 @@ Add the migration linter your ``INSTALLED_APPS``:
     ]
 
 
-``python manage.py lintmigrations [GIT_COMMIT_ID] [--ignore-name-contains=IGNORE_NAME_CONTAINS] [--include-apps INCLUDE_APPS [INCLUDE_APPS ...] | --exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]]``
+``python manage.py lintmigrations [GIT_COMMIT_ID] [--ignore-name-contains=IGNORE_NAME_CONTAINS] [--include-apps INCLUDE_APPS [INCLUDE_APPS ...] | --exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]] [--project-root-path=DJANGO_PROJECT_FOLDER``
```
Comment (David-Wobrock) [inline README.rst]: Same for consistency `[--project-root-path DJANGO_PROJECT_FOLDER]` (no `=` and missing closing `]`)

**Comment by mes3yd** [comment]
> On little comment, but generally I don't understand the need for this.
> How is it possible to run a Django management command without the `DJANGO_SETTINGS_MODULE` environment variable set (or using the default value present in your `manage.py`)?

I agree, `--settings` is a built-in option if you don't like environment variables and should be used instead of this.

File: README.rst (hunk 3 above)
Comment (linuxmaniac) [reply README.rst]: done

**Comment by linuxmaniac** [comment]
@dathoangse I would try to explain in more detail. If you have a django project like these:
```
/code/tmp
/code/tmp/manage.py
/code/tmp/settings
/code/tmp/settings/dev.py
/code/tmp/settings/__init__.py
/code/tmp/tmp/__init__.py
/code/tmp/tmp/wsgi.py
/code/tmp/tmp/urls.py
```
@mes3yd  notice that I will use --settings=settings.dev
```
./manage.py lintmigrations -v 2 --settings=settings.dev origin/master
INFO 2019-05-30 14:36:16,404 migration_linter 1 140364100678656 Executing cd /code/tmp/settings && git diff --relative --name-only --diff-filter=AR origin/master
INFO 2019-05-30 14:36:16,404 migration_linter 1 140364100678656 Executing cd /code/tmp/settings && git diff --relative --name-only --diff-filter=AR origin/master
```

``/code/tmp/settings`` is _NOT_ the project root and I need to be able to set it via params

File: README.rst [hunk 4]
```diff
@@ -43,12 +43,11 @@ Add the migration linter your ``INSTALLED_APPS``:
     ]
 
 
-``python manage.py lintmigrations [GIT_COMMIT_ID] [--ignore-name-contains=IGNORE_NAME_CONTAINS] [--include-apps INCLUDE_APPS [INCLUDE_APPS ...] | --exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]]``
+``python manage.py lintmigrations [GIT_COMMIT_ID] [--ignore-name-contains=IGNORE_NAME_CONTAINS] [--include-apps INCLUDE_APPS [INCLUDE_APPS ...] | --exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]] [--project-root-path DJANGO_PROJECT_FOLDER``
```
Comment (mrudelle) [inline README.rst]:
```suggestion
``python manage.py lintmigrations [GIT_COMMIT_ID] [--ignore-name-contains=IGNORE_NAME_CONTAINS] [--include-apps INCLUDE_APPS [INCLUDE_APPS ...] | --exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]] [--project-root-path DJANGO_PROJECT_FOLDER]``
```

**Comment by David-Wobrock** [comment]
I like the suggested solution in #64 and getting the root project path from the current working directory (using `os.getcwd()`).
However I don't think this will handle @linuxmaniac's problem, were the `settings` are not at the git root of the project. So I guess it is more flexible for all user's to have an option to specify the project root (in this context, equivalent to the git root).
Or am I missing something? :thinking: 

And I think multiple Django projects living inside a same git repo would be possible "by design" :thinking: If we allow specifying a project/git root, it will be the same for all projects. But the user will have to call the Django `lintmigrations` command once for every project anyway, and we match the migrations we find in git with the loaded Django migrations (so we stay in the scope of the project).
So your remark
> (or migration files could be intersected with the actual set of django migrations).

is what is happening actually happening more or less (see https://github.com/3YOURMIND/django-migration-linter/blob/master/django_migration_linter/migration_linter.py#L216 but the error case is badly handled currently, but easy to add)

**Comment by mrudelle** [comment]
@David-Wobrock  The settings folder (defined by `--settings`) and the root used to find migrations based on commit history can also live separately. 

Meaning that `_gather_migrations_git` would check the entire git repository. And filter out migrations from potential other projects with the MigrationLoader as you said. I think that would solve both @linuxmaniac and our problems.

@linuxmaniac Any plan of working on this PR in the future or should someone take it from here?

**Comment by linuxmaniac** [comment]
I just rebased and fixed the missing ``]`` at README.rst, thanks @mrudelle 
@David-Wobrock I'm missing what You want me to do, really

**Comment by David-Wobrock** [comment]
@linuxmaniac Thank you very much for rebasing.

I'm sorry if my message was confusing, it was more of a written thought on the different options we have to tackle this problem.
But you're patch is the best solution I could come up with. Merging it and releasing a new version :rocket:

**Pull request merged**